src/python/pants/backend/makeself/goals/run.py

```diff
@@ -0,0 +1,191 @@
+# Copyright 2024 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+import os
+from dataclasses import dataclass
+from typing import Optional, Tuple
+
+from pants.backend.makeself.target_types import (
+    MakeselfArchiveArgsField,
+    MakeselfArchiveFilesField,
+    MakeselfArchiveOutputPathField,
+    MakeselfArchivePackagesField,
+    MakeselfArchiveStartupScriptField,
+    MakeselfArthiveLabelField,
+)
+from pants.core.goals.package import BuiltPackage, PackageFieldSet
+from pants.core.goals.run import RunFieldSet, RunInSandboxBehavior, RunRequest
+from pants.core.util_rules.system_binaries import (
+    AwkBinary,
+    Base64Binary,
+    BasenameBinary,
+    BashBinary,
+    BinaryShims,
+    BinaryShimsRequest,
+    Bzip2Binary,
+    CatBinary,
+    CutBinary,
+    DateBinary,
+    DdBinary,
+    DfBinary,
+    DirnameBinary,
+    ExprBinary,
+    FindBinary,
+    GpgBinary,
+    GzipBinary,
+    HeadBinary,
+    IdBinary,
+    Md5sumBinary,
+    MkdirBinary,
+    PwdBinary,
+    RmBinary,
+    SedBinary,
+    ShasumBinary,
+    TailBinary,
+    TarBinary,
+    TestBinary,
+    WcBinary,
+    XzBinary,
+    ZstdBinary,
+)
+from pants.engine.fs import Digest
+from pants.engine.process import Process
+from pants.engine.rules import Get, collect_rules, rule
+from pants.util.logging import LogLevel
+
+
+@dataclass(frozen=True)
+class RunMakeselfArchive:
+    exe: str
+    input_digest: Digest
+    description: str
+    level: LogLevel = LogLevel.INFO
+    output_directory: Optional[str] = None
+    extra_args: Optional[Tuple[str, ...]] = None
+
+
+@rule(desc="Run makeself archive", level=LogLevel.DEBUG)
+async def run_makeself_archive(
+    request: RunMakeselfArchive,
+    awk: AwkBinary,
+    base64: Base64Binary,
+    basename: BasenameBinary,
+    bash: BashBinary,
+    bzip2: Bzip2Binary,
+    cat: CatBinary,
+    cut: CutBinary,
+    date: DateBinary,
+    dd: DdBinary,
+    df: DfBinary,
+    dirname: DirnameBinary,
+    expr: ExprBinary,
+    find: FindBinary,
+    gpg: GpgBinary,
+    gzip: GzipBinary,
+    head: HeadBinary,
+    id: IdBinary,
+    md5sum: Md5sumBinary,
+    mkdir: MkdirBinary,
+    pwd: PwdBinary,
+    rm: RmBinary,
+    sed: SedBinary,
+    shasum: ShasumBinary,
+    tail: TailBinary,
+    tar: TarBinary,
+    test: TestBinary,
+    wc: WcBinary,
+    xz: XzBinary,
+    zstd: ZstdBinary,
```

Comment:
I think this long list of required binaries is still problematic (e.g. `gpg` might not be installed... and I imagine people might be uninstalling `xz` recently!), and we don't seem to have wrapped up the conversation in https://github.com/pantsbuild/pants/pull/20360 . I'd like to at least agree on an approximate path forward before we land this.

Reply:
I think my comment got lost somewhere.

Yeah, I'm totally ok to remove the optional binaries and use mechanism from `adhoc_tool` in the next pr.